### PR TITLE
Add git call to find .git directory for git worktrees

### PR DESCRIPTION
**Pull request by Robert-Morgowicz**
Adds a new section to handleSubmoduleIfNeeded which queries git for the git dir if it's in a worktree instead of failing to find the config.  

- [x] All [tests](https://github.com/jfrog/jfrog-client-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/jfrog-client-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

closes https://github.com/jfrog/jfrog-cli/issues/1531

### CODE REVIEW

**Comment by Robert-Morgowicz** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by eyalbe4** [comment]
@Robert-Morgowicz,
I see that you're using `exec.Command("git", "rev-parse", "--git-dir")` as a fallback, but that brings the following question - 
What if the code used only this git command in the first place? Wouldn't it always work? Do we still need the code that runs before it? 
If not, will you be able to share more information about the two cases?

**Comment by eyalbe4** [comment]
Got it @Robert-Morgowicz.
In that case. I think we better have only the git command approach, and remove the older code. The code would be simpler and more predicable. We should also see iof there's a test that covers this, and if not, add the test.

**Comment by Robert-Morgowicz** [comment]
> Got it @Robert-Morgowicz. In that case. I think we better have only the git command approach, and remove the older code. The code would be simpler and more predicable. We should also see iof there's a test that covers this, and if not, add the test.

I reduced it to just using the new implementation (asking git for the location of the .git dir).  I tested it and it works for submodules as well.  I couldn't figure out how to add a test for git worktrees, though.  I see where to add one in `utils/vcsdetails_test.go`, but digging into `utils/tests/utils.go`, I don't see where the actual git calls used to create these test repos are coming from.

**Comment by eyalbe4** [comment]
Thank you @Robert-Morgowicz.
Here's how you can testing for the submodules functionality you added.
The `handleSubmoduleIfNeeded` function you improved is called from the `ReadConfig` function,
image

The `ReadConfig` function is tested by the `testReadConfig` function.
image

If you look at the first line of this test function, you'll see that it calls the `getDotGitPath(t)` function, which returns a path to a .git directory for the testing. This .git directory is copied to from the `testdata` directory.
You can take the same approach, and create a new `testReadConfigWithSubmodules` function, which worls similarly like the existing `testReadConfig` function. You can place your own submodules .git inside the tests data, and your new test function can copy and test it test it.

**Comment by lex-ibm** [comment]
recheck

**Comment by lex-ibm** [comment]
Hello @eyalbe4!

We've added a test case for worktrees. Is there anything else we need to do?

**Comment by lex-ibm** [comment]
Ping

**Comment by eyalbe4** [comment]
Let me review this soon @lex-ibm.

**Comment by notoriousUdo** [comment]
Any updates here?

File: utils/git.go [hunk 1]
```diff
@@ -73,38 +74,33 @@ func (m *GitManager) handleSubmoduleIfNeeded() {
...
 		return
 	}
-
-	// Extract path by removing prefix.
-	actualRelativePath := strings.TrimSpace(line[strings.Index(line, ":")+1:])
-	actualAbsPath := filepath.Join(filepath.Dir(m.path), actualRelativePath)
-	exists, err = fileutils.IsDirExists(actualAbsPath, false)
+	ResolvedGitPath := strings.TrimSpace(stdout.String())
+	// trim the worktree path to just the
+	// worktreePathReduced := filepath.Dir(filepath.Dir(worktreePath))
```
Comment (eyalbe4) [inline utils/git.go:92]: The above two commented lines seem unclear / unnecessary.

File: utils/git.go [hunk 2]
```diff
@@ -73,38 +74,33 @@ func (m *GitManager) handleSubmoduleIfNeeded() {
...
-		m.err = errorutils.CheckErrorf("failed to parse .git path for submodule")
+		m.err = err
 		return
 	}
-
-	// Extract path by removing prefix.
-	actualRelativePath := strings.TrimSpace(line[strings.Index(line, ":")+1:])
-	actualAbsPath := filepath.Join(filepath.Dir(m.path), actualRelativePath)
-	exists, err = fileutils.IsDirExists(actualAbsPath, false)
+	ResolvedGitPath := strings.TrimSpace(stdout.String())
```
Comment (eyalbe4) [inline utils/git.go:90]: ResolvedGitPath --> resolvedGitPath

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: utils/git.go (hunk 2 above)
Comment (lex-ibm) [reply utils/git.go]: Refactored that variable name

**Comment by lex-ibm** [comment]
Hey @eyalbe4! 
Happy holidays!

**Comment by schmidtl25** [comment]
Hi @eyalbe4 - can this PR be re-reviewed and merged?

**Pull request merged**